### PR TITLE
Fix: make apiserver work behind a feature toggle

### PR DESCRIPTION
**Pull request by zserge**
After several changes on main the existing apiserver feature got broken. This PR restores the functionality:

* It registers apiserver in the background services, to make it instantiated by wire (until dskit/modules are restored)
* It respects the feature toggle and does not start the apiserver if disabled
* It fixes the creation of apiserver certificates (another dskit module that got broken?)
* Backports admission plugins to make it work with namespace-scoped resources

I can't say I understand half of the code there, but it works.

### CODE REVIEW

**Review by joanlopez — APPROVED** [review]
> I can't say I understand half of the code there..

More or less the same here, but it looks good AFAICT!

**Review by mildwonkey — APPROVED** [review]
👍🏻 I'm ok with merging this as-is for now. I'm not sure what the longer term plan for this is (@charandas @toddtreece might know) but we deal with that later.

**Review by charandas — APPROVED** [review]
I just pushed one change removing some no-op code which we left on our feature branch as a placeholder. No reason, the test admission plugins should land in main.

**Comment by charandas** [comment]
Trying to resolve these conflicts now.

**Comment by charandas** [comment]
This PR and #74177 are incompatible. A lot of those wiresets are necessary for grafana-apiserver, certgenerator and the like. If we can chat 1:1, I can try to better understand what we are trying to do here. Then I would be much more able to help unblock this PR.

**Comment by charandas** [comment]
On taking another look, seems the branch named [mildwonkey/server-target-cmd](https://github.com/grafana/grafana/tree/mildwonkey/server-target-cmd) is doing away with wiresets intentionally. Since some of the work from that branch has landed in main, maybe best to wait to enable the GrafanaAPIServer until all of it has landed?

File: pkg/services/grafana-apiserver/service.go
Comment (charandas) [inline pkg/services/grafana-apiserver/service.go:145]: Ideally, we would be able to enable this handler. Its part of our desired APIServer routing that we want to expose through Grafana.

However, `RoutingRegister` is not liking being injected into ModulesServer which is where we `New` the GrafanaAPIServer Service. This is the error I am getting.

```shell
wire: /Users/charandas/go/src/github.com/grafana/grafana/pkg/server/wire.go:430:1: inject InitializeModuleServer: no provider found for github.com/grafana/grafana/pkg/api/routing.RouteRegister
        needed by *github.com/grafana/grafana/pkg/server.ModuleServer in provider set "wireExtsModuleServerSet" (/Users/charandas/go/src/github.com/grafana/grafana/pkg/server/wireexts_oss.go:125:31)
```

For now, I disabled it and left a TODO.

**Comment by charandas** [comment]
GrafanaAPIServer starts successfully now provided `custom.ini` has the following:

```
app_mode = development
target = grafana-apiserver

[feature_toggles]
grafanaAPIServer = true
```

That's a lot of gates in my opinion but whatever it takes to not break the workflow of existing developers.

File: pkg/services/grafana-apiserver/service.go
Comment (mildwonkey) [reply pkg/services/grafana-apiserver/service.go:145]: @charandas that error means that you need to add the RoutingRegister dependency to the OSS `wireExtsModuleServerSet` (it's already in the enterprise one). Adding that as a dependency may pull in more dependencies, though, and at the moment that means you get to find and fix them _one at a time_ (took me 2 days to get enterprise right ☹️ )

To continue the conversation in slack about wire vs. wireexts, there are a couple of things to know (if I'm explaining too much, I'm sorry, I'm trying to get all my thoughts in one place):

- Our dependencies are initialized with a function (usually `ProvideService`) 
- Those functions may return an interface or a concrete implementation. 
- Returning an interface lets us bind different implementations depending on if we're building Grafana OSS or enterprise

wire.go has the "main" or "general" or maybe "generic" - pick your insufficiently accurate term - wiresets with most of Grafana's dependencies, and the wireexts_* files are for OSS vs. enterprise implementations.

When ProvideService returns a concrete implementation, all you need to add to wire.go is the initializer method name: `thing.ProvideService`

But if `thing.ProvideService` returns an interface, you need to include a `wire.Bind` directive to bind the concrete implementation:

```
thing.ProvideService, // returns a Service interface
wire.Bind(new(thing.Service)), new(thingimplementation.ServiceImplementation) // binds a Service implementation
```

When there's a different implementation for OSS vs. enterprise, those bindings go into the appropriate wireexts_*.go files. Build flags (and some symlinking) control which wireset is built.

**Review by mildwonkey — APPROVED** [review]

File: pkg/services/grafana-apiserver/service.go
Comment (charandas) [reply pkg/services/grafana-apiserver/service.go:145]: Mostly got a crash course in wire since joining Grafana and understand it enough, have used it as well. But the separation between various wiresets within Grafana - I could benefit from getting a better understanding of.

**Pull request merged**
